Pilot does not init k8s config store in k8s environment when istio discovery service flag --registries=Mock is set.

**Bug description**
Istio discovery service flag `--registries=Mock` is set, pilot does not init k8s config store in k8s environment .
This flag just indicates the platform service registries to read from, not config store.
In some cases, in stead of specifying config source or file path for ConfigController i think pilot should use k8s as default config source in k8s environment. 

The description of flag` --registries` is : https://istio.io/latest/docs/reference/commands/pilot-discovery/ .
Comma separated list of platform service registries to read from (choose one or more from {Kubernetes, Mock}) (default [Kubernetes])



[ ] Docs
[ ] Installation
[ ] Networking
[X ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Pilot should init k8s config store in k8s environment when istio discovery service flag `--registries=Mock` is set.

**Steps to reproduce the bug**
1.Download install package from https://github.com/istio/istio/releases/tag/1.8.0-alpha.2.
2. Add --registries=Mock into file: manifests/charts/istio-control/istio-discovery/templates/deployment.yaml.
```yaml
...         
         args:
          - "discovery"
          - --registries=Mock
          - --monitoringAddr=:15014
{{- if .Values.global.logging.level }}
          - --log_output_level={{ .Values.global.logging.level }}
{{- end}}
{{- if .Values.global.logAsJson }}
          - --log_as_json
{{- end }}
          - --domain
          - {{ .Values.global.proxy.clusterDomain }}
{{- if .Values.global.oneNamespace }}
          - "-a"
          - {{ .Release.Namespace }}
{{- end }}
{{- if .Values.pilot.plugins }}
          - --plugins={{ .Values.pilot.plugins }}
{{- end }}
          - --keepaliveMaxServerConnectionAge
          - "{{ .Values.pilot.keepaliveMaxServerConnectionAge }}"
...          
```
3.Use istioctl to install pilot.
```shell
istioctl install --charts=./manifests --set profile=minimal -y
```
4. See istod deployment args, `--registries=Mock` is added.
```yaml
...   
   spec:
      containers:
      - args:
        - discovery
        - --registries=Mock
        - --monitoringAddr=:15014
        - --log_output_level=default:info
        - --domain
        - cluster.local
        - --keepaliveMaxServerConnectionAge
        - 30m
...        
```
5.Pilot has errors as follows:
```
2020-11-19T15:19:51.562648Z	info	version: 1.8.0-alpha.2-ec349ef2a8a0bc846a87c8c10e02511e4865c9cc-Clean
2020-11-19T15:19:51.563031Z	info	flags:
2020-11-19T15:19:51.563453Z	info	initializing mesh networks
2020-11-19T15:19:51.563930Z	info	mesh networks configuration: {
   "networks": {
   }
}
2020-11-19T15:19:51.564032Z	info	initializing mesh handlers
2020-11-19T15:19:51.564058Z	info	initializing controllers
2020-11-19T15:19:51.564068Z	info	No certificates specified, skipping K8S DNS certificate controller
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x108 pc=0x240ad43]

goroutine 1 [running]:
istio.io/istio/pilot/pkg/config/kube/crdclient.New(0x0, 0x0, 0xc000078009, 0x7, 0x0, 0x0, 0x0, 0x0, 0xdf8475800, 0x7ffc6575c86f, ...)
	istio.io/istio/pilot/pkg/config/kube/crdclient/client.go:142 +0x123
istio.io/istio/pilot/pkg/bootstrap.(*Server).makeKubeConfigController(0xc001282ea0, 0xc00143e780, 0x7, 0x7fd677a08e98, 0x2, 0x8)
	istio.io/istio/pilot/pkg/bootstrap/configcontroller.go:428 +0xb8
istio.io/istio/pilot/pkg/bootstrap.(*Server).initK8SConfigStore(0xc001282ea0, 0xc00143e780, 0xb26800, 0xc00013e5a0)
	istio.io/istio/pilot/pkg/bootstrap/configcontroller.go:132 +0x50
istio.io/istio/pilot/pkg/bootstrap.(*Server).initConfigController(0xc001282ea0, 0xc00143e780, 0x0, 0x0)
	istio.io/istio/pilot/pkg/bootstrap/configcontroller.go:84 +0x993
istio.io/istio/pilot/pkg/bootstrap.(*Server).initControllers(0xc001282ea0, 0xc00143e780, 0x17, 0xc00181b778)
	istio.io/istio/pilot/pkg/bootstrap/server.go:1030 +0x11e
istio.io/istio/pilot/pkg/bootstrap.NewServer(0xc00143e780, 0x0, 0xc00144a420, 0x0)
	istio.io/istio/pilot/pkg/bootstrap/server.go:242 +0x8ca
main.glob..func1(0x4d471c0, 0xc000a59340, 0x0, 0x7, 0x0, 0x0)
	istio.io/istio/pilot/cmd/pilot-discovery/main.go:70 +0x175
github.com/spf13/cobra.(*Command).execute(0x4d471c0, 0xc000a592d0, 0x7, 0x7, 0x4d471c0, 0xc000a592d0)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x4d46f20, 0x0, 0x0, 0x0)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	istio.io/istio/pilot/cmd/pilot-discovery/main.go:198 +0x31
```

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
```shell
$ istioctl version --remote
client version: 1.8.0-alpha.2
control plane version: 1.8.0-alpha.2
data plane version: 1.8.0-alpha.2 (6 proxies)
```

**How was Istio installed?**
```shell
istioctl install --charts=./manifests --set profile=minimal -y
```

--- a/pilot/pkg/bootstrap/server.go
+++ b/pilot/pkg/bootstrap/server.go
@@ -476,7 +476,7 @@ func (s *Server) initKubeClient(args *PilotArgs) error {
 					hasK8SConfigStore = true
 				}
 			}
-		}  else if args.RegistryOptions.KubeConfig != "" {
+		} else if args.RegistryOptions.KubeConfig != "" {
 			hasK8SConfigStore = true
 		}
 	}

--- a/pilot/pkg/bootstrap/server.go
+++ b/pilot/pkg/bootstrap/server.go
@@ -468,11 +468,16 @@ func (s *Server) initKubeClient(args *PilotArgs) error {
 			if err != nil {
 				return fmt.Errorf("failed reading mesh config: %v", err)
 			}
+			if len(meshConfig.ConfigSources) == 0 && args.RegistryOptions.KubeConfig != "" {
+				hasK8SConfigStore = true
+			}
 			for _, cs := range meshConfig.ConfigSources {
 				if cs.Address == "k8s://" {
 					hasK8SConfigStore = true
 				}
 			}
+		}  else if args.RegistryOptions.KubeConfig != "" {
+			hasK8SConfigStore = true
 		}
 	}
 

--- a/pilot/pkg/serviceregistry/mock/discovery.go
+++ b/pilot/pkg/serviceregistry/mock/discovery.go
@@ -245,7 +245,7 @@ func (sd *ServiceDiscovery) GetIstioServiceAccounts(svc *model.Service, ports []
 
 func (sd *ServiceDiscovery) NetworkGateways() map[string][]*model.Gateway {
 	// TODO use logic from kube controller if needed
-	panic("implement me")
+	return map[string][]*model.Gateway{}
 }
 
 type Controller struct{}

